src/backend/distributed/planner/multi_planner.c

```diff
@@ -295,6 +295,15 @@ CreateDistributedPlan(PlannedStmt *localPlan, Query *originalQuery, Query *query
 	resultPlan = FinalizePlan(localPlan, distributedPlan);
 
 	/*
+	 * PostgreSQL checks access permissions in standard_ExecutorStart(), but
+	 * we do these checks here just before returning from planner. After this
+	 * point, we don't have access to our local tables in our plan. Therefore,
```

Comment:
I think `local tables` is a bit confusing since we're using distributed relations in the next sentence. I'd rather use `original queries' range table list` or smth like that.